src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -172,6 +187,102 @@ void validateCmekKey(FailureCollector failureCollector, Map<String, String> argu
     validateCmekKeyLocation(cmekKeyName, null, location, failureCollector);
   }
 
+  /**
+   * Validates job label key value pairs, as per the following rules:
+   * Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes.
+   * Defined in the following link:
+   * <a href="https://cloud.google.com/bigquery/docs/adding-labels#job-label">Docs</a>
+   * @param failureCollector failure collector
+   */
+  void validateJobLabelKeyValue(FailureCollector failureCollector) {
+    Set<String> reservedKeys = BigQueryUtil.BQ_JOB_LABEL_SYSTEM_KEYS;
+    int maxLabels = 64 - reservedKeys.size();
+    int maxKeyLength = 63;
+    int maxValueLength = 63;
+
+    String validLabelRegex = "[a-z0-9_-]+";
```

Comment:
This regex does not look correct for keys, according to [doc](https://cloud.google.com/bigquery/docs/labels-intro#requirements), `Keys must start with a lowercase letter or international character.` is not satisfied.